airbyte-integrations/connectors/destination-mariadb-columnstore/src/main/java/io/airbyte/integrations/destination/mariadb_columnstore/MariadbColumnstoreSqlOperations.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.mariadb_columnstore;
+
+import com.vdurmont.semver4j.Semver;
+import io.airbyte.db.jdbc.JdbcDatabase;
+import io.airbyte.integrations.base.JavaBaseConstants;
+import io.airbyte.integrations.destination.jdbc.JdbcSqlOperations;
+import io.airbyte.protocol.models.AirbyteRecordMessage;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MariadbColumnstoreSqlOperations extends JdbcSqlOperations {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MariadbColumnstoreSqlOperations.class);
+  private final String MINIMUM_VERSION = "5.5.3";
+  Pattern VERSION_PATTERN = Pattern.compile("^(\\d+\\.\\d+\\.\\d+)-MariaDB");
+  private boolean isLocalFileEnabled = false;
+
+  @Override
+  public void insertRecordsInternal(final JdbcDatabase database,
+                                    final List<AirbyteRecordMessage> records,
+                                    final String schemaName,
+                                    final String tmpTableName)
+      throws SQLException {
+    if (records.isEmpty()) {
+      return;
+    }
+
+    verifyLocalFileEnabled(database);
+
+    File tmpFile = null;
+    Exception primaryException = null;
+    try {
+      tmpFile = Files.createTempFile(tmpTableName + "-", ".tmp").toFile();
+      writeBatchToFile(tmpFile, records);
+
+      final String query = String.format(
+          "LOAD DATA LOCAL INFILE %s INTO TABLE %s.%s FIELDS TERMINATED BY ',' ENCLOSED BY '\"' ESCAPED BY '\\\"' LINES TERMINATED BY '\\r\\n'",
+          String.format("'%s'", tmpFile.getAbsolutePath()), schemaName, tmpTableName);
+
+      database.execute(query);
```

Comment:
nit: I understood that you mimicked  our current mysql / postgres load strategy that is using temp localfile. I wondered if you had considered an (in memory?) solution that avoids relying on available disk space.

Reply:
Yes, another solution is using [ColumnStore Bulk write SDK](https://mariadb.com/kb/en/columnstore-bulk-write-sdk/#getting-started-with-java). Using the SDK, it can loads data from in memory. And this method enables  fast data ingestion similar to 'LOAD DATA LOCAL INFILE' using [cpimport](https://mariadb.com/kb/en/columnstore-data-ingestion/). 
But, since [the SDK doesn't updated since 2018](https://github.com/mariadb-corporation/mariadb-columnstore-api), I could not have confidence to use it.